lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -1497,7 +1497,7 @@ public Parser() {
     }
 
     public Parser(final NameValidator validate) {
-      this.validate = validate;
+      this.validate = Objects.requireNonNull(validate, "validate");
```

Comment:
How about ?
```suggestion
      this.validate = validate != null ? validate : NameValidator.NO_VALIDATION);
```
Schema.parse() uses `NameValidator.NO_VALIDATION` if there is no need to validate.
I guess you use `Parser.parse()` directly and that's why you see NPEs later.

Reply:
Ah, that makes much more sense. Can't agree more.